### PR TITLE
gdag.d: minor refactor of aewalk()

### PR DESCRIPTION
**Pull request by WalterBright**
Includes turning `aeclear()` into a nested function.

### CODE REVIEW

File: src/dmd/backend/gdag.d [hunk 1]
```diff
@@ -79,15 +79,15 @@ void builddags()
     assert(dfo);
     flowae();                       /* compute available expressions */
     if (go.exptop <= 1)             /* if no AEs                     */
-            return;
+        return;
     aetype = Aetype.cse;
 
     debug
-        for (uint i = 0; i < go.exptop; i++)
+        foreach (i, e; go.expnod[0 .. go.exptop])
```
Comment (thewilsonator) [inline src/dmd/backend/gdag.d]: `i` is unused

File: src/dmd/backend/gdag.d [hunk 2]
```diff
@@ -101,9 +101,8 @@ void builddags()
     {
         /* This is the 'correct' algorithm for CSEs. We can't use it    */
         /* till we fix the code generator.                              */
-        for (uint i = 0; i < dfotop; i++)
+        foreach (i, b; dfo[0 .. dfotop])
```
Comment (thewilsonator) [inline src/dmd/backend/gdag.d]: ditto

**Review by thewilsonator — APPROVED** [review]
Otherwise LGTM.

File: src/dmd/backend/gdag.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/backend/gdag.d]: line 88

File: src/dmd/backend/gdag.d (hunk 2 above)
Comment (WalterBright) [reply src/dmd/backend/gdag.d]: line 110

File: src/dmd/backend/gdag.d (hunk 1 above)
Comment (thewilsonator) [reply src/dmd/backend/gdag.d]: Fair enough.

**Pull request merged**